### PR TITLE
Mimo docu

### LINKED ISSUE #5: In-tree top-level documentation
I think about the proposed MIMO feature not just as a functioning tool but as well as an instructive and attractive feature for GNU Radio beginners to understand MIMO. I am therefore planning to additionally write a little in-tree top-level documentation, dropping some knowledge about the different MIMO techniques.

### LINKED ISSUE #9: In-tree top-level documentation
I think about the proposed MIMO feature not just as a functioning tool but as well as an instructive and attractive feature for GNU Radio beginners to understand MIMO. I am therefore planning to additionally write a little in-tree top-level documentation, dropping some knowledge about the different MIMO techniques.

### LINKED ISSUE #22: In-tree top-level documentation
I think about the proposed MIMO feature not just as a functioning tool but as well as an instructive and attractive feature for GNU Radio beginners to understand MIMO. I am therefore planning to additionally write a little in-tree top-level documentation, dropping some knowledge about the different MIMO techniques.

### LINKED ISSUE #23: In-tree top-level documentation
I think about the proposed MIMO feature not just as a functioning tool but as well as an instructive and attractive feature for GNU Radio beginners to understand MIMO. I am therefore planning to additionally write a little in-tree top-level documentation, dropping some knowledge about the different MIMO techniques.

### PR DESCRIPTION
**Pull request by MLsmd**
Write in-tree top-level documentation about the implemented MIMO basics. Includes:

- Introduction and Motivation
- Mathematical description
- Receiver diversity and diversity combining which closes #5 
- STBC's and Alamouti's Code which closes #9 
- Differential STBC which closes #22 
- V-BLAST which closes #23

### CODE REVIEW

File: docs/doxygen/other/mimo.dox [hunk 1]
```diff
@@ -11,20 +11,79 @@
...
+Each reveiving antenna has its independend Additive White Gauissian Noise (AWGN) applied. We will neglect this noise vector at some points, if it is not relevant for the calculations.
```
Comment (evaristesome) [inline docs/doxygen/other/mimo.dox]: Each receiving antenna has its independent Additive White Gaussian Noise (AWGN) applied. We will neglect this noise vector at some points, if it is not relevant for the calculations.

File: docs/doxygen/other/mimo.dox [hunk 2]
```diff
@@ -11,20 +11,79 @@
...
+For the theoretical introduction to MIMO, the reader does not require any preknowledge about MIMO itselve, but a fundamental background in communications theory is assumed. Although, if you are not yet familiar with the basic functionalities of GNU Radio, you should probably stop here and come back after you read an introduction to GNU Radio and did the tutorials. \ref page_exploring_gnuradio is also a good way to start.
 
 \subsection mimo_motivation Motivation
 
-\section mimo_math_intro Mathematical Descritption of MIMO
+\section mimo_math_intro Mathematical Descritption
+
+Before we dive right into the many great MIMO techniques and algorithms, we first have to find a smart and consistent way to describe our communications systems that is now expanded to multiple antennas at the transmitter and receiver. For this purpose, we use the great power of linear algebra.
 
-Before we dive right into the many great MIMO techniques and algorithms, we first have to find a smart and consistent way to describe our communications systems that is now expanded to multiple antennas at the transmitter and receiver. For this purpose, we use the great power of linear algebrea.
```
Comment (evaristesome) [inline docs/doxygen/other/mimo.dox]: Before we dive right into the many great MIMO techniques and algorithms, we first have to find a smart and consistent way to describe our communications systems that is now expanded to multiple antennas at the transmitter and receiver. For this purpose, we use the great power of linear algebra.

File: docs/doxygen/other/mimo.dox [hunk 3]
```diff
@@ -11,20 +11,79 @@
...
-Let us assume a MIMO system with M transmitting antennas and N receiving antennas and let's further assume a discrete time. One complex sample is precessed at each time slot and antenna, meaning it is modulated and transmitted or received and sampled. Hence, the transmitted data per time slot can be described in an M-dimensinal vector, where each dimension represents one of the transmitting antennas. In equal way, we can describe the received data in an N-dimensional vector. A Discrete-Time AWGN channel with slow and frequency flat fading satisfies for the purpose of a general explanation of fundamental MIMO algorithms. Each transmission between one transmitting antenna and a receiving antenna is characterized through a complex path coefficient where the attenuation is described by the magnitude and the applied phase shift by the phase of this path coefficient. All in all, there exist NxM different path coefficients. They can be collected in a channel matrix H with the dimensions NxM. A transmission through this channel equals a matrix multiplication of the channel matrix H with the transmission vector S, resulting in the reception vector R.
```
Comment (evaristesome) [inline docs/doxygen/other/mimo.dox]: Let us assume a MIMO system with M transmitting antennas and N receiving antennas and let's further assume a discrete time. One complex sample is processed at each time slot and antenna, meaning it is modulated and transmitted or received and sampled. Hence, the transmitted data per time slot can be described in an M-dimensinal vector, where each dimension represents one of the transmitting antennas. In equal way, we can describe the received data in an N-dimensional vector. A Discrete-Time AWGN channel with slow and frequency flat fading satisfies for the purpose of a general explanation of fundamental MIMO algorithms. Each transmission between one transmitting antenna and a receiving antenna is characterized through a complex path coefficient where the attenuation is described by the magnitude and the applied phase shift by the phase of this path coefficient. All in all, there exist NxM different path coefficients. They can be collected in a channel matrix H with the dimensions NxM. A transmission through this channel equals a matrix multiplication of the channel matrix H with the transmission vector S, resulting in the reception vector R.

File: docs/doxygen/other/mimo.dox [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+# or any later version published by the Free Software Foundation;
+# with no Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.
+# A copy of the license is included in the section entitled "GNU
+# Free Documentation License".
+
+/*! \page page_mimo MIMO
+
+\section mimo_introduction Introduction
+
+The following manual includes both, an introduction to the theoretical background
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: Comma before "both" is superfluous, I think.

File: docs/doxygen/other/mimo.dox [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+
+/*! \page page_mimo MIMO
+
+\section mimo_introduction Introduction
+
+The following manual includes both, an introduction to the theoretical background 
+of MIMO and a practical usage manual of all the GNU Radio blocks which are included 
+in the section MIMO.
+
+For the theoretical introduction to MIMO, the reader does not require any preknowledge
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: "preknowledge" -> prior knowledge

File: docs/doxygen/other/mimo.dox [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+/*! \page page_mimo MIMO
+
+\section mimo_introduction Introduction
+
+The following manual includes both, an introduction to the theoretical background 
+of MIMO and a practical usage manual of all the GNU Radio blocks which are included 
+in the section MIMO.
+
+For the theoretical introduction to MIMO, the reader does not require any preknowledge 
+about MIMO itselve, but a fundamental background in communications theory is assumed.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: itselve -> itself

File: docs/doxygen/other/mimo.dox [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+GNU Radio and did the tutorials. \ref page_exploring_gnuradio is also a good way to start.
+
+\subsection mimo_motivation Motivation
+
+The use of multiple antennas at the transmitter and receiver (MIMO) is an essential method in
+wireless communications. It uses space as an additional degree of freedom in form of spatially
+separated antennas at the transmitter and receiver to significantly improve the performance of
+a communications system. <b>Spatial diversity</b> can be used to mitigate the effect of fading by
+combining independently fading signal paths at the receiver [1], thereby achieving an increased
+signal-to-noise ratio (SNR). <b>Spatial multiplexing</b> obtains independent spatial paths at the
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: obtains -> uses

File: docs/doxygen/other/mimo.dox [hunk 8]
```diff
@@ -0,0 +1,420 @@
...
+same frequency and time which is used to transmit parallel data streams on these paths, thereby
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: which is used -> x (delete)

File: docs/doxygen/other/mimo.dox [hunk 9]
```diff
@@ -0,0 +1,420 @@
...
+increasing the channel capacity by a multiple factor.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: by a multiple factor -> x (This is true only under certain circumstances, anyway.)

File: docs/doxygen/other/mimo.dox [hunk 10]
```diff
@@ -0,0 +1,420 @@
...
+
+A lot of effort was put in the last few decades into the development and research of algorithms
+which could exploit the mentioned effects as efficiently and effectively as possible. The goal is to
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: could -> x

File: docs/doxygen/other/mimo.dox [hunk 11]
```diff
@@ -0,0 +1,420 @@
...
+converge to the theoretical Shannon channel capacity without disregarding the complexity and
+feasibility for practical use cases.
+
+MIMO is not a standard application in GNU Radio yet. Although there are several projects
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: This is basic GNU Radio documentation. As awesome as your GSoC project is, this becomes in-tree code and is therefore no "project". The code you write is supposed to become the standard way of dealing with MIMO applications in GNU Radio.

File: docs/doxygen/other/mimo.dox [hunk 12]
```diff
@@ -0,0 +1,420 @@
...
+many projects. As a demonstration of how to integrate the MIMO capability into a transmission
+system, the second part of this project includes the realized MIMO algorithms into the existing
+OFDM transceiver of GNU Radio.
+
+\section mimo_math_intro Mathematical Description
+
+Before we dive right into the many great MIMO techniques and algorithms, 
+we first have to find a smart and consistent way to describe our communications systems 
+that is now expanded to multiple antennas at the transmitter and receiver. 
+For this purpose, we use the great power of linear algebra.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: This is the very neutral scientific me speaking: "the many great MIMO techniques" and "the great power of linear algebra" feels a little too much to me.

File: docs/doxygen/other/mimo.dox [hunk 13]
```diff
@@ -0,0 +1,420 @@
...
+
+\f$ y \f$ describes a sample that results out of a combination of N received 
+input signals \f$ r \f$, each weighted by a complex coefficient \f$ \alpha_i \f$.
+
+\subsection mimo_selection_combining Selection Combining (SC)
+
+One of the easiest combining methods is simply selecting the path with the best 
+input signal in terms of SNR as output signal. When assuming equal noise at each receiver, 
+this is equal to a selection of the path coefficient with the highest magnitude. 
+This leads to the following combining algorithm (if there are no equal csi coefficients):
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: csi -> CSI

File: docs/doxygen/other/mimo.dox [hunk 14]
```diff
@@ -0,0 +1,420 @@
...
+the weighting vector is calculated with the new data. The idea is, 
+that a stream tag comes with each new symbol, where symbol means a frame of data samples which belong together. 
+The definition of a symbol (and therefore the frequency of tagging CSI) is the job of the system designer who uses this block.
+
+\subsection mimo_diversity_combining_csi Providing CSI
+
+Providing channel state information (CSI) via stream tags is a fundamental 
+requirement for the diversity combining block and also for many other MIMO blocks 
+which are introduced later on. The channel can be estimated at the receiver with pilot symbols, for example. 
+A detailled instruction to channel estimation for MIMO channels is not part of
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: detailled -> detailed

File: docs/doxygen/other/mimo.dox [hunk 15]
```diff
@@ -0,0 +1,420 @@
...
+
+Providing channel state information (CSI) via stream tags is a fundamental 
+requirement for the diversity combining block and also for many other MIMO blocks 
+which are introduced later on. The channel can be estimated at the receiver with pilot symbols, for example. 
+A detailled instruction to channel estimation for MIMO channels is not part of 
+this documentation but you can have an inpirational look at the channel estimation 
+of the MIMO-OFDM transceiver [not yet implemented; scheduled for August 2018].
+
+The produced CSI is a channel matrix of the dimensions NxM. In case of diversity combining, 
+the matrix reduces to a N-dimensional vector.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: Why N-dimensional? I would have expected that after diversity combining, it's M coefficients (for the different transmit antennas). Probably I'm confusing things here, but maybe clarify this.

File: docs/doxygen/other/mimo.dox [hunk 16]
```diff
@@ -0,0 +1,420 @@
...
+// Append stream tags with CSI to data stream.
+add_item_tag(0, nitems_written(0) + tag_pos, pmt::mp("csi"), csi_pmt);
+\endcode
+
+\section mimo_stbc Space Time Block Code (STBC)
+
+Using a Multiple Input Multiple Output (MIMO) system adds the new dimension space (via multiple antennas) 
+to the pool of  degrees of freedom, next to frequency, time, code, polarization, etc., 
+that a system designer can use. This gives the possibility to distribute the signal 
+not just over time, but over on other (or multiple) dimension.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: over on other -> fix this

File: docs/doxygen/other/mimo.dox [hunk 17]
```diff
@@ -0,0 +1,420 @@
...
+
+Using a Multiple Input Multiple Output (MIMO) system adds the new dimension space (via multiple antennas) 
+to the pool of  degrees of freedom, next to frequency, time, code, polarization, etc., 
+that a system designer can use. This gives the possibility to distribute the signal 
+not just over time, but over on other (or multiple) dimension. 
+We all know this technique from Frequency Division Multiplexing (FDM), 
+where the data is split into multiple streams which are simultaneously transmitted over time, 
+each on a different sub-carrier.
+
+The same principle underlays Space Time Codes (STCs), where the data is distributed over time and space.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: underlays -> underlies

File: docs/doxygen/other/mimo.dox [hunk 18]
```diff
@@ -0,0 +1,420 @@
...
+
+\f[
+\mathbf{r} = \mathbf{Hs} + \mathbf{n} \quad \text{with} \; \mathbf{s} \in \mathbb{C}^M,\; 
+\mathbf{r} \in \mathbb{C}^N,\; \mathbf{H} \in \mathbb{C}^{N\times M},\; \mathbf{n} \in \mathbb{C}^N
+\f]
+
+for a MIMO system with M transmitting antennas and N antennas at the receiver side. 
+To describe a STBC, we have to add the dimension time to the scheme. 
+Thanks to the block character of STBCs, the time dimension is limited to the block length T. 
+This would results in
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: would -> x

File: docs/doxygen/other/mimo.dox [hunk 19]
```diff
@@ -0,0 +1,420 @@
...
+
+\f[
+\mathbf{R} = \mathbf{HS} + \mathbf{N} \quad \text{with} \; \mathbf{S} \in \mathbb{C}^{M\times T},\; 
+\mathbf{R} \in \mathbb{C}^{N\times T},\; \mathbf{H} \in \mathbb{C}^{N\times M\times T},\; \mathbf{N} \in \mathbb{C}^{N\times T}
+\f]
+
+Fortunately, we don’t have to deal with 3-dimensional matrices. 
+Instead we can pass each column of S separately through the channel, 
+and therefore again calculating each time step separately. However,
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: calculating -> calculate

File: docs/doxygen/other/mimo.dox [hunk 20]
```diff
@@ -0,0 +1,420 @@
...
+This is provided via stream tags (see \ref mimo_diversity_combining_csi) and 
+handled in the same way as for the \ref mimo_diversity_combining.
+
+\section mimo_differential_stbc Differential STBC
+
+As in SISO systems, a differential modulation makes the presence of CSI at the receiver redundant. 
+The price we are paying for this enormous simplification of the receiver is a loss of approximately 3dB in SNR.
+
+The algorithm of this implementation follows [4]. 
+Basically this is the Alamouti STBC which is expanded about its differential feature.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: This doesn't sound good.

File: docs/doxygen/other/mimo.dox [hunk 21]
```diff
@@ -0,0 +1,420 @@
...
+\section mimo_differential_stbc Differential STBC
+
+As in SISO systems, a differential modulation makes the presence of CSI at the receiver redundant. 
+The price we are paying for this enormous simplification of the receiver is a loss of approximately 3dB in SNR.
+
+The algorithm of this implementation follows [4]. 
+Basically this is the Alamouti STBC which is expanded about its differential feature. 
+The rough idea of this algorithm is as follows:
+
+- Transform each input sequences (= a pair of complex input samples) from the
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: sequences -> sequence

File: docs/doxygen/other/mimo.dox [hunk 22]
```diff
@@ -0,0 +1,420 @@
...
+The input sequences can be identified by two weighting coefficients A and B 
+which describe the proportion of each basis vector to the sequence. A and B 
+can be calculated over a dot product of the input sequence with the basis vectors.
+- Differentially encode the transformed input sequences and distribute the resulting 
+output sequence among the 2 antennas. The differential operation can be done 
+by a complex multiplication.
+- Use Alamouti’s scheme to produce an orthogonal sequence to the generated output 
+sequence which is appended to the output sequence. Summarizing, one 
+input sequence of 2 complex samples results in one output sequence of 2 complex samples 
+at each of the 2 antennas, respectively. This equals a code rate of r=1.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: Wasn't it capital R for the code rate before?

File: docs/doxygen/other/mimo.dox [hunk 23]
```diff
@@ -0,0 +1,420 @@
...
+
+\subsection mimo_differential_stbc_implementation Implementational Thoughts
+
+Two fundamental design questions have shaped the differential STBC blocks:
+
+- <b>Which type of input samples are valid?</b>\n
+→ Generally, the algorithm qualifies a complex PSK constellation of any 
+constellation order as valid input. The encoder does not even has to know the modulation order,
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: has -> have

File: docs/doxygen/other/mimo.dox [hunk 24]
```diff
@@ -0,0 +1,420 @@
...
+at each of the 2 antennas, respectively. This equals a code rate of r=1.
+
+\subsection mimo_differential_stbc_implementation Implementational Thoughts
+
+Two fundamental design questions have shaped the differential STBC blocks:
+
+- <b>Which type of input samples are valid?</b>\n
+→ Generally, the algorithm qualifies a complex PSK constellation of any 
+constellation order as valid input. The encoder does not even has to know the modulation order, 
+because it has now effect on the algorithm.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: now -> no

File: docs/doxygen/other/mimo.dox [hunk 25]
```diff
@@ -0,0 +1,420 @@
...
+To encode a sequence, we need the current input sequence as well as the 
+previously encoded sequence. We are therefore producing one sequence for each 
+consumed sequence (that is a sync block after definition), but we need memory 
+of one sequence. This predecessor is defined for all sequences except the very 
+first one. For this first sequence, we define a random ‘dummy’ sequence (in this 
+implementation this sequence equals the defined basis vector). 
+This makes the encoder practically a sync block although the very first 
+transmitted sequence does not contain any useful information.
+The situation for the decoder is data stream wise similar to the one of the encoder. 
+Despite, I implemented the decoder with a general block. This has to reasons:\n
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: I wouldn't use "I" here. Use passive.

File: docs/doxygen/other/mimo.dox [hunk 26]
```diff
@@ -0,0 +1,420 @@
...
+1.) The use of a ‘dummy’ sequence at the beginning of a reception leads 
+to a wrong decoded first sequence. We don’t want to produce false data at the decoder.\n
+2.) A fully fleshed receiver includes much more than just a MIMO receiver.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: fully fleshed -> fully fledged. You probably confused this with fleshed out, which also has a somewhat different meaning.

File: docs/doxygen/other/mimo.dox [hunk 27]
```diff
@@ -0,0 +1,420 @@
...
+implementation this sequence equals the defined basis vector). 
+This makes the encoder practically a sync block although the very first 
+transmitted sequence does not contain any useful information.
+The situation for the decoder is data stream wise similar to the one of the encoder. 
+Despite, I implemented the decoder with a general block. This has to reasons:\n
+1.) The use of a ‘dummy’ sequence at the beginning of a reception leads 
+to a wrong decoded first sequence. We don’t want to produce false data at the decoder.\n
+2.) A fully fleshed receiver includes much more than just a MIMO receiver. 
+There are synchronizers for time and frequency synchronization and the data is 
+maybe structured in different data blocks. This leads to very many situations
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: very many -> a lot of (or just many)

File: docs/doxygen/other/mimo.dox [hunk 28]
```diff
@@ -0,0 +1,420 @@
...
+transmitted sequence does not contain any useful information.
+The situation for the decoder is data stream wise similar to the one of the encoder. 
+Despite, I implemented the decoder with a general block. This has to reasons:\n
+1.) The use of a ‘dummy’ sequence at the beginning of a reception leads 
+to a wrong decoded first sequence. We don’t want to produce false data at the decoder.\n
+2.) A fully fleshed receiver includes much more than just a MIMO receiver. 
+There are synchronizers for time and frequency synchronization and the data is 
+maybe structured in different data blocks. This leads to very many situations 
+that are equal to a new beginning of a transmission and therefore to the argument of 1.).\n
+To handle this issue, I introduced stream tags to the decoder block,
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: again, no first-person speech please.

File: docs/doxygen/other/mimo.dox [hunk 29]
```diff
@@ -0,0 +1,420 @@
...
+only storing the sequence in his memory. For the following sequences of the block, 
+the decoder works data synchronously.
+
+\subsection mimo_differential_stbc_usage Usage
+
+The differential STBC does not require any CSI. The usage of this MIMO scheme is 
+therefore as simple as inserting the gr::digital::diff_stbc_encoder_cc after the 
+symbol mapping at the transmitter and the gr::digital::diff_stbc_decoder_cc 
+before the symbol demapper at the receiver.
+To avoid the reception of erroneous sample, the usage of 'start' tags is highly recommended.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: This is unclear.

**Review by fwunsch — CHANGES_REQUESTED** [review]
Nice docs you wrote there! Some minor comments, though. The only major (structural) thing is, that I would like you to think about this docs as the in-tree docs for a GNU Radio module and less as a documentation of your project. This is supposed to become the standard way of dealing with MIMO applications in GNU Radio and that should also be the mindset when writing this.

Good job!

File: docs/doxygen/other/mimo.dox [hunk 30]
```diff
@@ -392,6 +392,55 @@ To avoid the reception of erroneous sample, the usage of 'start' tags is highly
 For example, the tags can be set in the synchronizer, 
 when it gets on track and produces synchronized samples.
 
+\section mimo_mux Spatial Multiplexing
+
+So far we used the mulitple antennas of a MIMO system to achieve diversity and therefore decreasing the error rate of the communication system. An other possibilty for the exploitation of the mulitple antennas is spatial multiplexing. The premise is the transmission of M independent symbols per time slot over M transmit antennas, therefore increasing the data rate instead of the transmission quality. An easy way to do so offers parallel encoding which simply demultiplexes the data stream into M parallel streams. One of the Bell Labs Layered Space Time (BLAST) architectures does exactly that: Encoding serial data into a vertical vector which gives the algorithm its name: Vertical-BLAST or short V-BLAST [1].
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: typo: multiple

File: docs/doxygen/other/mimo.dox [hunk 31]
```diff
@@ -392,6 +392,55 @@ To avoid the reception of erroneous sample, the usage of 'start' tags is highly
...
+- <b>H must be quadratic</b>: At this time, we did not state any relationship between M and N, yet. At first we can observe, that N ≥ M must be fulfilled to gain enough linear equations to solve the system. Most textbooks solve this issue by introducing the Moore-Penrose inverse (or pseudo inverse) which doesn’t require a quadratic matrix.
+\f[
+\mathbf{W} = (\mathbf{H^*H)^{-1}H^*}
+\f]
+The calculation of the pseudo inverse \f$ W \f$ has a more complex calculation complexity. Instead of one inversion, it requires 2 adjunctions, 1 inversion and 2 more matrix multiplications. To avoid this increase of decoder complexity, the decoder calculates the inverse instead of the pseudo-inverse, therefore setting the number of receive antennas fixed to N=M. This, however, is not a restriction to the V-BLAST algorithm in general. For a NxM MIMO setup with  N > M, the maximum increase of the data rate is min{M, N}. This data rate is limited by M for N ≥ M. The remaining N-M transmitter antennas are used to apply a diversity gain. This means for our system, that we have to reduce the number of streams at the receiver from N to M, by combining some streams with the diversity combining block.
```
Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: "more complex" -> higher

Comment (fwunsch) [inline docs/doxygen/other/mimo.dox]: Is your SNR calculated per antenna or is it assumed to be the same for all RX antennas? If the latter is the case, combining some streams may introduce an SNR inequality between the receive streams which is then suboptimal for MMSE equalization.

**Review by fwunsch — COMMENTED** [review]
Again, nicely written docs and only very few comments from my side. I'd like to merge this as soon as you have finished describing the MMSE equalizer (what you state as TODO at the end of the document) to get rid of your unfinished milestones. The OFDM integration can be added later and pulled in separately.

File: docs/doxygen/other/mimo.dox (hunk 31 above)
Comment (MLsmd) [reply docs/doxygen/other/mimo.dox]: The MMSE equalizer in the V-BLAST decoder considers an individual SNR estimation for each incoming path. It is therefore no problem to combine multiple paths in advance if the SNR information of the combined paths is updated afterwards.

**Review by fwunsch — APPROVED** [review]

**Pull request merged**